C/services/common/notification_subscription.cpp

```diff
@@ -242,11 +241,9 @@ void NotificationSubscription::unregisterSubscription(const string& assetName)
 	string callBackURL = api->getCallBackURL();
 
 	// Get all NotificationSubscriptions
-	m_subscriptionMutex.lock();
 	std:map<std::string, std::vector<SubscriptionElement>>&
```

Comment:
Is the assumption this routine is called when you are already holding the subscription lock? If so please state this clearly in the comments that document the entry point.